### DISCUSSION TITLE
Installing homebrew pdfsandwich on Mac OS

### DISCUSSION THREAD

**Discussion by tifflaujoa** [Homebrew/discussions#3395] [Getting started]
Hello, 

My main goal of installing pdfsandwich is to extract data from an image-based PDF into a spreadsheet. However, I've been struggling to install it through the terminal. 
This is what I've typed so far on the command line: 
ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)" < /dev/null 2> /dev/null

Homebrew has been installed successfully, but somehow I can't get pdfsandwich installed when I typed : brew install pdfsandwich

Another question is where do I find the command lines for the conversion from pdf to spreadsheet?

Any tips or advice would be appreciated.

**Comment by carlocab** [comment]
Does calling `brew` work? What is the output of `brew config` and `brew doctor`?

> Another question is where do I find the command lines for the conversion from pdf to spreadsheet?

You'll need to consult the pdfsandwich documentation for this. See http://www.tobias-elze.de/pdfsandwich/. If there is no documentation there then maybe you can try calling `--help` or checking the manpages.

**Comment by tifflaujoa** [comment]
**This is what I'm getting as output for brew doctor:**
_Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  sdl_

Output for brew config:
tiffanylaujoa@Tiffanys-Air ~ % brew config
HOMEBREW_VERSION: 3.5.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: 2b01354f2f8cdbf1315dce4f063b54cdecf95663
Last commit: 21 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 53b24f6e16b8859b5d318e5fa861795b951950f0
Core tap last commit: 9 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.1.6 build 1316
Git: 2.32.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.4-arm64
CLT: 13.4.0.0.1.1651278267
Xcode: N/A
Rosetta 2: false

Now for pdfsandwich, I did go over the documentation, and I tried the following commands, and none of them give me any results. 

_tiffanylaujoa@Tiffanys-Air ~ % pdfsandwich -coo /Users/tiffanylaujoa/Downloads/elections.pdf
Fatal error: exception Failure("Could not open file ")_
_tiffanylaujoa@Tiffanys-Air ~ % pdf-renderer /Users/tiffanylaujoa/Downloads/elections.pdf_
zsh: command not found: pdf-renderer
tiffanylaujoa@Tiffanys-Air ~ % pdfsandwich help
Fatal error: exception Failure("Could not open file help")
_tiffanylaujoa@Tiffanys-Air ~ % pdfsandwich -convert elections.pdf
Fatal error: exception Failure("Could not find program elections.pdf. Make sure this program exists and can be found in your search path.\nUse command line options to specify a custom binary.")_

Do you know by any chance what a custom binary mean?

**Reply by gromgit** [reply]
You're not getting the results you expect because you're not using `pdfsandwich` correctly. This is not a Homebrew problem; you need to read the documentation quite a bit more closely, and keep in mind that computers follow the principle of **DWYTNWYM** ("Did What You Typed, Not What You Meant").
>```
>pdfsandwich -coo /Users/tiffanylaujoa/Downloads/elections.pdf
>```

The `-coo` example in the documentation looks like this:
```
pdfsandwich -coo "-deskew 40%" scanned_file.pdf
```
and the option's help text says:
```
  -coo 		 -coo options : additional convert options; make sure to quote;
		  e.g. -coo "-normalize -black-threshold 75%"
		  call convert --help or man convert for all convert options
```
You're clearly missing an argument to `-coo`.

>```
>pdf-renderer /Users/tiffanylaujoa/Downloads/elections.pdf
>```

Where did `pdf-renderer` come from?

>```
>pdfsandwich help
>```

The documentation states:
```
  -help  Display this list of options
  --help  Display this list of options
```
You're missing at least one important punctuation character.

>```
>pdfsandwich -convert elections.pdf
>```

The `-convert` option takes an argument, which is the name of a program that behaves in a manner similar to the `convert` program of the ImageMagick software suite:
```
  -convert 	 -convert filename : name of convert binary (default: convert)
```

>Do you know by any chance what a custom binary mean?

Based on the documentation, it just means "program that behaves in a manner similar to the `convert` program of the ImageMagick software suite, for when you don't have ImageMagick installed".

And since `imagemagick` should already be installed (it's one of the dependencies of `pdfsandwich`), it's highly unlikely that you need to use the `-convert` option.

**Comment by gromgit** [comment]
> Another question is where do I find the command lines for the conversion from pdf to spreadsheet?

Try searching online for `convert pdf to spreadsheet`. `pdfsandwich` is clearly _not_ what you need for this task, since the documentation describes its purpose as follows:

> pdfsandwich generates "sandwich" OCR pdf files, i.e. pdf files which contain only images (no text) will be processed by optical character recognition (OCR) and the text will be added to each page invisibly "behind" the images.

so its output is an annotated PDF, not a spreadsheet in any format.
